lib/hMap.ml

```diff
@@ -242,11 +242,18 @@ struct
     try
       let m = Int.Map.find h s in
       let m = Map.add k x m in
-      Int.Map.update h m s
+      Int.Map.update_simple h m s
     with Not_found ->
       let m = Map.singleton k x in
       Int.Map.add h m s
 
+  (* when Coq requires OCaml 4.06 or later, the module type
+     CSig.MapS may include the signature of OCaml's "update",
+     requiring an implementation here, which could be just:
+
+       let update k f s = assert false (* not implemented *)
```

Comment:
I don't understand why we would want to do that.

Reply:
You mean, other than provide an implementation? 

An implementation of `update` a la OCaml is a bit complicated.